lib/segment/src/vector_storage/chunked_utils.rs

```diff
@@ -0,0 +1,92 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use memmap2::MmapMut;
+
+use crate::common::mmap_ops::{create_and_ensure_length, open_write_mmap, write_to_mmap};
+use crate::entry::entry_point::{OperationError, OperationResult};
+
+const STATUS_FILE_NAME: &str = "status.json";
+
+const MMAP_CHUNKS_PATTERN_START: &str = "chunk_";
+const MMAP_CHUNKS_PATTERN_END: &str = ".mmap";
+
+pub struct MmapStatus {
+    pub len: usize,
```

Comment:
Maybe we can add here information about flush status (is there any non-flushed changes)